lib/uplink/uplink.go

```diff
@@ -112,14 +127,28 @@ func (u *Uplink) OpenProject(ctx context.Context, satelliteAddr string, apiKey A
 		return nil, err
 	}
 
-	// TODO: we shouldn't need segment or stream stores to manage buckets
-	segments := segments.NewSegmentStore(metainfo, nil, eestream.RedundancyStrategy{}, maxBucketMetaSize.Int(), maxBucketMetaSize.Int64())
-	streams, err := streams.NewStreamStore(segments, maxBucketMetaSize.Int64(), nil, 0, storj.Unencrypted)
+	// TODO: we shouldn't need segment or stream stores to manage buckets.
+	// We won't make any use of them. But you can't skimp on any of this
+	// setup, or things will break as they stand now.
+	encryptionKeyThatWeWontEvenUse := &storj.Key{}
```

Comment:
We have to initialize the stream store with a meaningful encryption key. Otherwise, it won't be able to decrypt the bucket metainfo for buckets that were not created with libuplink. This is one of the reasons for failing some of the gateway tests.

We can resolve this issue by having an addition `EncKey` field to `uplink.Config.Volatile`.